### PR TITLE
Add linting target

### PR DESCRIPTION
**Pull request by richardcase**
### Description
A new make target has been added to run gometalinter.
There are a lot of linting errors!!

### Checklist
- [x] Code compiles correctly (i.e `make build`)
- [ ] Added tests that cover your change (if possible)
- [x] All tests passing (i.e. `make test`)
- [ ] Added/modified documentation as required (such as the README)
- [ ] Added yourself to the `humans.txt` file

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -19,6 +19,19 @@ test: generate
 	@go test -v -covermode=count -coverprofile=coverage.out ./pkg/... ./cmd/...
 	@test -z $(COVERALLS_TOKEN) || goveralls -coverprofile=coverage.out -service=circle-ci
 
+.PHONY: lint
+lint:
+	gometalinter --exclude=^vendor\/  ./...
+
+.PHONY: install-lint-deps
+install-lint-deps:
+	curl https://git.io/vp6lP | sh
```
Comment (errordeveloper) [inline Makefile]: Could you please add `github.com/alecthomas/gometalinter` to https://github.com/weaveworks/eksctl/blob/master/build/Gopkg.toml and https://github.com/weaveworks/eksctl/blob/master/build/install.sh instead? That way all the dependencies will be included in the build image, or installed locally via `make install-build-deps`.

Comment (richardcase) [reply Makefile]: Seems like and easy thing to do but been struggling getting this working. I must be doing something stupid!

Comment (errordeveloper) [reply Makefile]: happy to pick this up, would be nice to include in 0.1.2!

**Comment by errordeveloper** [comment]
I made it fail only on errors, we should probably change it long-term, for the purpose of this PR, it seems right to not let it make all our builds fail.

**Comment by richardcase** [comment]
Great @errordeveloper  - thanks for fixing this.

**Pull request merged**